File: Daily/sbgo/Leetcode633-平方数之和.go
package sbgo

func judgeSquareSum(c int) bool {
	a, b := 0, 0
	for a <= b {
		if c > a*a+b*b {
			b++
		} else if c < a*a+b*b {
			a++
			b--
		} else {
			return true
		}
	}
	return false
}

File: Leetcode/PY/Leetcode633-平方数之和.py
from math import sqrt
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        if c == 0:
            return True
        i = 0
        while i * i <= c:
            b = sqrt(c - i * i)
            if b == int(b):
                return True
            i += 1
        return False